app/features/syllabus_generator/document_loaders.py

```diff
@@ -105,167 +105,210 @@ def load(self, url):
         return documents
     
 def load_pdf_documents(pdf_url: str, verbose=False):
-    pdf_loader = FileHandler(PyPDFLoader, "pdf")
-    docs = pdf_loader.load(pdf_url)
-
-    if docs:
-        split_docs = splitter.split_documents(docs)
-
-        if verbose:
-            logger.info(f"Found PDF file")
-            logger.info(f"Splitting documents into {len(split_docs)} chunks")
+    try:
+        pdf_loader = FileHandler(PyPDFLoader, "pdf")
+        docs = pdf_loader.load(pdf_url)
 
-        full_content = [doc.page_content for doc in split_docs]
-        full_content = " ".join(full_content)
+        if docs:
+            split_docs = splitter.split_documents(docs)
+            if verbose:
+                logger.info(f"Found PDF file")
+                logger.info(f"Splitting documents into {len(split_docs)} chunks")
 
-        return full_content
-        
+            full_content = " ".join([doc.page_content for doc in split_docs])
+            return full_content
+    
+    except FileHandlerError as e:
+        logger.error(f"File handler error: {str(e)}")
+        raise
+    except Exception as e:
+        logger.error(f"Unexpected error: {str(e)}")
+        raise DocumentLoadError("Unexpected error while loading PDF", pdf_url) from e
 
 def load_csv_documents(csv_url: str, verbose=False):
-    csv_loader = FileHandler(CSVLoader, "csv")
-    docs = csv_loader.load(csv_url)
+    try:
```

Comment:
Yes, I understand. The try-catch blocks are indeed redundant here since the FileHandler class already handles those errors. I'll remove the try-catch blocks from all the remaining document loaders as well. Thank you for pointing that out.